src/app/payment-status/payment-status.component.ts

```diff
@@ -9,28 +12,38 @@ import { StorageService } from '../storage.service';
   templateUrl: './payment-status.component.html',
   styleUrls: ['./payment-status.component.scss'],
 })
-export class PaymentStatusComponent {
+export class PaymentStatusComponent implements OnDestroy {
+  private subscription: Subscription;
   /**
    * Payment status response binding.
    */
   statusResponse: string;
 
-  constructor(private storageService: StorageService) {
+  constructor(
+    private apiService: RightfootApiService,
+    private storageService: StorageService
+  ) {
     this.statusResponse = JSON.stringify(
       JSON.parse(this.storageService.getStoredPlaidResponse()),
       null,
       2
     );
   }
 
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
   /**
-   * Set payment status response from the storage.
+   * Retrieves payment status
    */
-  checkStatus() {
-    this.statusResponse = JSON.stringify(
-      JSON.parse(this.storageService.getStoredPlaidResponse()),
-      null,
-      2
-    );
+  public checkStatus() {
+    this.subscription = this.storageService.storedPaymentUuid$
+      .pipe(switchMap((uuid) => this.apiService.getPayment(uuid)))
+      .subscribe((payment) => {
+        this.statusResponse = JSON.stringify(JSON.parse(payment.toString()), null, 2);
+      });
```

Comment:
Instead of managing a subscription, I'd say it is better to use `take(1)` in the pipe after `switchMap`. Your code will unsubscribe right after a request completes.

Reply:
done 67fb20f